tests/ban/ban_manager_test.php

```diff
@@ -0,0 +1,686 @@
+				[
+					'item'		=> '*@foo.bar',
+					'end'		=> '0',
+					'reason'	=> '1',
+					'mode'		=> 'email',
+				],
+			],
+		];
+	}
+
+	/**
+	 * @dataProvider data_check_ban
+	 */
+	public function test_check_ban($user_data, $expected)
+	{
+		$this->assertEquals($expected, $this->ban_manager->check($user_data));
+	}
+
+	public function data_get_bans(): array
+	{
+		return [
+			[
+				'foo',
+				'',
+				type_not_found_exception::class
+			],
+			[
+				'ip',
+				[
+					[
+						'ban_id' => '6',
+						'ban_userid' => 0,
+						'ban_item' => '10.0.0.1/28',
+						'ban_start' => '1111',
+						'ban_end' => '0',
+						'ban_reason' => 'HAHAHA',
+						'ban_reason_display' => '1',
+						'ban_mode' => 'ip',
+					],
+					[
+						'ban_id' => '2',
+						'ban_userid' => 0,
+						'ban_item' => '127.0.0.1',
+						'ban_start' => '1111',
+						'ban_end' => '0',
+						'ban_reason' => 'HAHAHA',
+						'ban_reason_display' => '1',
+						'ban_mode' => 'ip',
+					],
+					[
+						'ban_id' => '3',
+						'ban_userid' => 0,
+						'ban_item' => '127.1.1.1',
+						'ban_start' => '1111',
+						'ban_end' => '0',
+						'ban_reason' => 'HAHAHA',
+						'ban_reason_display' => '1',
+						'ban_mode' => 'ip',
+					],
+					[
+						'ban_id' => '7',
+						'ban_userid' => 0,
+						'ban_item' => '2001:4860:4860::8888/12',
+						'ban_start' => '1111',
+						'ban_end' => '0',
+						'ban_reason' => 'HAHAHA',
+						'ban_reason_display' => '1',
+						'ban_mode' => 'ip',
+					],
+				],
+			],
+			[
+				'email',
+				[
+					[
+						'ban_id' => '9',
+						'ban_userid' => 0,
+						'ban_item' => '*@foo.bar',
+						'ban_start' => '1111',
+						'ban_end' => '0',
+						'ban_reason' => 'HAHAHA',
+						'ban_reason_display' => '1',
+						'ban_mode' => 'email',
+					],
+					[
+						'ban_id' => '5',
+						'ban_userid' => 0,
+						'ban_item' => 'bar@example.org',
+						'ban_start' => '1111',
+						'ban_end' => '0',
+						'ban_reason' => 'HAHAHA',
+						'ban_reason_display' => '1',
+						'ban_mode' => 'email',
+					],
+				],
+			],
+			[
+				'user',
+				[
+					[
+						'ban_id' => '4',
+						'ban_item' => '4',
+						'ban_start' => '1111',
+						'ban_end' => '0',
+						'ban_reason' => 'HAHAHA',
+						'ban_reason_display' => '1',
+						'ban_mode' => 'user',
+						'ban_userid' => 4,
+						'user_id' => '4',
+						'username' => '',
+						'username_clean' => 'ipv6_user',
+					],
+				],
+			],
+		];
+	}
+
+	/**
+	 * @dataProvider data_get_bans
+	 */
+	public function test_get_bans($ban_type, $expected, $expected_exception = false)
+	{
+		if ($expected_exception !== false)
+		{
+			$this->expectException($expected_exception);
+		}
+
+		// Fix weird sorting by postgres
+		if (strtolower(substr(PHP_OS, 0, 3)) === 'linux' && is_array($expected) && $ban_type == 'email' && $this->db->get_sql_layer() == 'postgres')
+		{
+			$expected = array_reverse($expected);
+		}
+
+		$this->assertEquals($expected, $this->ban_manager->get_bans($ban_type));
```

Comment:
Something like this fixes the test.

```suggestion
		$actual = $this->ban_manager->get_bans($ban_type);

		// Sort both arrays by ban_item to be synced
		if (is_array($expected) && !empty($actual))
		{
			usort($expected, function($a, $b)
				{
					return strcmp($a['ban_item'], $b['ban_item']) <=> 0;
				}
			);

			usort($actual, function($a, $b)
				{
					return strcmp($a['ban_item'], $b['ban_item']) <=> 0;
				}
			);
		}

		$this->assertEquals($expected, $actual);
```

Reply:
Thanks, that should solve it.